src/types.ts

```diff
@@ -113,6 +117,22 @@ export interface PredefinedType extends BaseValueType {
   name: string;
 }
 
+export interface LiteralType extends BaseValueType {
+  kind: ValueTypeKind.literalType;
+  type:
+    | {
+        kind: ValueTypeKind.basicType;
```

Comment:
Why we cannot use `BasicType` directly here

Reply:
I reuse the code from parseLiteralNode.

Reply:
Look like only `value` in `BasicType` is changed to `type`.  Do you think it will take much effort to use `BasicType` in `parseLiteralNode`?

Reply:
The parseLiteralNode also uses enumType, so maybe it's okay to keep it for now.